integration-test/src/main/resources/commandrunner/qa-command-runner.py

```diff
@@ -0,0 +1,120 @@
+    results = list()
+    for c in config:
+        result = {}
+        description = c["description"]
+        command = c["command"]
+        code, out, err = run_command(command)
+        result["command"] = c["command"]
+        result["description"] = c["description"]
+        result["code"] = code
+        result["out"] = out.rstrip() if out else out
+        result["err"] = err.rstrip() if err else err
+        results.append(result)
+    print(json.dumps(results))
+
+
+def execute_salt_commands(command_file_location, config, hosts, host_groups):
+    global SALT_CMD_PREFIX
+    ping_out_file = "/tmp/qa_test_ping.json"
+    test_cmd = "%s '*' test.ping --out=json --out-file=%s --out-indent=-1" % (SALT_CMD_PREFIX, ping_out_file)
+    available_nodes = []
+    test_code, _, _ = run_command(test_cmd)
+    test_json = read_json_line_file(ping_out_file)
+    if test_json:
+        for key in test_json:
+            if str(test_json[key]) == "True":
+                available_nodes.append(key)
+    qa_path = '/srv/salt/qa'
+    if os.path.exists('/srv/salt') and not os.path.exists(qa_path):
+        os.mkdir(qa_path)
+    script_file = os.path.abspath(__file__)
+    script_file_basename = os.path.basename(script_file)
+    command_file_basename = os.path.basename(command_file_location)
+    try:
+        shutil.copy(command_file_location, os.path.join(qa_path, command_file_basename))
+    except shutil.SameFileError:
+        pass
+    try:
+        shutil.copy(script_file, os.path.join(qa_path, script_file_basename))
+    except shutil.SameFileError:
+        pass
+    available_nodes_str = ",".join(available_nodes)
+    cmd_copy_script_file = "%s -L %s cp.get_file salt:///qa/%s /tmp/%s" % (
+    SALT_CMD_PREFIX, available_nodes_str, script_file_basename, script_file_basename)
+    cmd_copy_command_file = "%s -L %s cp.get_file salt:///qa/%s /tmp/%s" % (
+    SALT_CMD_PREFIX, available_nodes_str, command_file_basename, command_file_basename)
+    run_command(cmd_copy_script_file)
+    run_command(cmd_copy_command_file)
+    final_output = '/tmp/qa_test_results.json'
+    run_all_commands_cmd = \
+        "%s -L %s cmd.run 'python3 /tmp/%s -c /tmp/%s -l' --out=json --out-file=%s --out-indent=-1" % (
+    SALT_CMD_PREFIX, available_nodes_str, script_file_basename, command_file_basename, final_output)
+    run_command(run_all_commands_cmd)
+    final_json = read_json_line_file(final_output)
+    outputs = {}
+    if final_json:
+        for key in final_json:
+            outputs[key] = json.loads(final_json[key])
+    print(json.dumps(outputs))
+
+
+if __name__ == "__main__":
+    parser = optparse.OptionParser("usage: %prog [options]")
+    parser.add_option("-c", "--config", dest="config", default=None, type="string",
+                      help="Configuration file for executing salt commands.")
+    parser.add_option("--hosts", dest="hosts", default=None, type="string", help="Comma separated hosts filter,")
```

Comment:
note these options are not covered in the script yet, sorry for not spending more time on it to finish this script.
also would make sense to include one more layer for the output which would should the result of the command itself (in order to avoid that if the command fails with error code 1, it returns as an exception for the ssh client), but actually that is good for start

Reply:
im fine with this for now will continue to changes but we need to make sure the e2e test framework will be able to parse the new response

Reply:
Created 2 follow-up JIRAs:
- [CB-18257 [API E2E] QA Command Runner script improvements with command options](https://jira.cloudera.com/browse/CB-18257)
- [CB-18253 [API E2E] QA Command Runner script need to be exit based on the executed commands exit codes](https://jira.cloudera.com/browse/CB-18253)